src/main/java/mate/academy/rickandmorty/controller/CharactersController.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.rickandmorty.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.CharacterDto;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Characters management", description = "Endpoints to managing character")
+@RestController
+@RequestMapping(value = "/characters")
+@RequiredArgsConstructor
+public class CharactersController {
+    private final CharacterService characterService;
+
+    @GetMapping("/random")
+    @Operation(summary = "Get random Character",
+            description = "Get one random Character from DB")
+    public CharacterDto getAll() {
+        return characterService.getRandom();
+    }
+
+    @GetMapping("/by_name")
```

Comment:
I'd remove url suffix.
Filtered search is like a prolongation of getAll method that should be available without suffix.
Agree, that it can be a bit confusing because you don't have getAll method. But getAll is not required by the task.
And I don't think this is the reason to change endpoints naming convention
```suggestion
    @GetMapping
```